### ISSUE TITLE
Fly login for non existing team should not result in login successful message in WebUI

### ISSUE THREAD

**Issue by aliculPix4D** [concourse/concourse#7519]
## Summary

Running fly login and setting a non-existing team results in "login successful" in the WebUI. This can cause confusion to the users. At the end, fly command fails as expected with `error: team 'foo' does not exist`.

## Steps to reproduce

Running fly login and setting a non-existing team results in "login successful" in the WebUI. At the end, fly command fails as expected with `error: team 'foo' does not exist`.

`fly -t target-foo login --team-name=foo --concourse-url=https://concourse.ci.com --open-browser`
or even
`fly -t target-foo login --team-name=foo --open-browser`
if the target already exists.

## Expected results

WebUI should already inform us that the team does not exists and that login will fail because of non existing team. No token should be transferred to fly.

## Actual results
In the WebUI we get the following message:

```
login successful!

your token has been transferred to fly.

you may now close this window.

```
command line output returns:

```

fly -t target-foo login --team-name=foo --open-browser

logging in to team 'foo'

navigate to the following URL in your browser:

  https://concourse.ci.com/login?fly_port=42153

or enter token manually (input hidden): 
error: team 'foo' does not exist
```

## Additional context

![login](https://user-images.githubusercontent.com/50741304/132365625-3a4bb16d-a240-4cac-9324-c1756b602edf.png)

## Triaging info

* Concourse version: v7.3.1
* Browser (if applicable): Google Chrome Version 93.0.4577.63
* Did this used to work? Unknown

**Comment by taylorsilva** [comment]
I don't think this is a bug, though I do agree with your aspirations for a better UX when a user is logging in via fly.

The reason I don't think this is a bug is because of how `fly` and the web server think about auth, they think differently about it.
- fly: auth info is a target + specific team
- web: auth is about the individual user

Let's go over the auth flow from fly -> web -> fly

You do `fly login` with a specific team. Fly redirects you to a URL of this format: `https://example.com/login?fly_port=42153`. From the URL one can see that the only info fly is passing onto the web sever is `fly_port`. The web server knows nothing about what team you're logging into. The web server doesn't care if you belong to zero teams, 50 teams, one team; teams don't matter to the current auth flow process. For example, anyone with a github account can log into [https://ci.concourse-ci.org/](https://ci.concourse-ci.org/), even if they belong to zero teams.

The basic auth screen does not ask for a team:
<img width="381" alt="image" src="https://user-images.githubusercontent.com/4583837/132400820-fab2a27e-871c-41cb-a91b-bcc2f60d03fc.png">

So the web server never knows what team you're logging into. Fly knows and does do that final check, a you showed. Fly will print an error if the team does not exist.
```
error: team 'foo' does not exist
```

To have the web server perform this check, the team the user is trying to log into will need to be passed to the web server. This is what leads me to see this as a feature/enhancement to Concourse, and not a bug fix. There's nothing to fix, only more code to add.

**Comment by aliculPix4D** [comment]
@taylorsilva Thank you for your detailed response. Sure it make sense that this is not a bug, but a enhancement.

**Comment by taylorsilva** [comment]
# Issue Bankruptcy

This issue is being closed as part of 2025's **Issue Bankruptcy** declaration. [See here for details](https://blog.concourse-ci.org/posts/2025-01-06-concourse-2024-review/)

If you think this issue still has merit then feel free to open a new issue and reference this one.
